akka-sse/src/main/scala/de/heikoseeberger/akkasse/ServerSentEventParser.scala

```diff
@@ -34,23 +34,35 @@ private object ServerSentEventParser {
 
   private val linePattern = """([^:]+): ?(.*)""".r
 
-  private def parseServerSentEvent(lines: Vector[String]): ServerSentEvent = {
-    val values = lines.foldLeft(Map.empty[String, Vector[String]].withDefault(_ => Vector.empty)) {
-      case (m, "")                                                      => m
-      case (m, linePattern(field @ (Data | Event | Id | Retry), value)) => m.updated(field, m(field) :+ value)
-      case (m, field)                                                   => m.updated(field, m(field) :+ "")
+  private def parseFields(lines: Vector[String]) = {
+    var data = Vector.empty[String]
+    var eventType = null: String
+    var id = null: String
+    var retry = null: String
+    for (line <- lines) {
+      line match {
+        case linePattern(field @ (Data | Event | Id | Retry), value) => field match {
```

Comment:
Great idea, will do that. Yet in practice the effect will be close to zero, because fields without value (without a trailing colon) are an exotic corner case.